File: Selenium/Project/AlchemyLMSProject.java
package LMS;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.util.List;

public class AlchemyLMSProject {
    WebDriver driver;
    @BeforeMethod
    public void launchapplication() {
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/lms");
    }
    @Test
    public void Activity1() {
        String sTitle = driver.getTitle();
        Assert.assertEquals("Alchemy LMS – An LMS Application", sTitle);

    }
    @Test
    public void Activity2() {
        String sHeadingText = driver.findElement(By.tagName("h1")).getText();
        Assert.assertEquals("Learn from Industry Experts", sHeadingText);

    }
    @Test
    public void Activity3() {
        List<WebElement> sTitles = driver.findElements(By.xpath("//h3[@class='uagb-ifb-title'][1]"));
        int i=1;
        for (WebElement webElement : sTitles) {
            if(i==1) {
                System.out.println(webElement.getText());
                Assert.assertEquals("Actionable Training", webElement.getText());
            }
            i++;
        }

    }

    @Test
    public void Activity4() {
        List<WebElement> sTitles = driver.findElements(By.xpath("//h3[@class='entry-title']"));
        int i=1;
        for (WebElement webElement : sTitles) {
            if(i==2) {
                System.out.println(webElement.getText());
                Assert.assertEquals("Email Marketing Strategies", webElement.getText());
            }
            i++;
        }

    }

    @Test
    public void Activity5() {
        driver.findElement(By.linkText("My Account")).click();
        String sTitle = driver.getTitle();
        Assert.assertEquals("My Account – Alchemy LMS", sTitle);
    }

    @Test
    public void Activity6() {
        driver.findElement(By.linkText("My Account")).click();
        String sTitle = driver.getTitle();
        Assert.assertEquals("My Account – Alchemy LMS", sTitle);
        driver.findElement(By.linkText("Login")).click();
        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();
        WebElement logoutbutton = driver.findElement(By.linkText("Logout"));
        Assert.assertTrue(logoutbutton.isDisplayed());

    }

    @Test
    public void Activity7() {
        driver.findElement(By.linkText("All Courses")).click();
        List<WebElement> sTitles = driver.findElements(By.xpath("//h3[@class='entry-title']"));
        System.out.println("Total Number of course is: "+sTitles.size());
        for (WebElement webElement : sTitles) {
                System.out.println(webElement.getText());

        }
    }

    @Test
    public void Activity8() {
        driver.findElement(By.linkText("My Account")).click();
        String sTitle = driver.getTitle();
        Assert.assertEquals("My Account – Alchemy LMS", sTitle);
        driver.findElement(By.linkText("Login")).click();
        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();
        WebElement logoutbutton = driver.findElement(By.linkText("Logout"));
        Assert.assertTrue(logoutbutton.isDisplayed());

    }
    @Test
    public void Activity9() {
        driver.findElement(By.linkText("My Account")).click();
        String sTitle = driver.getTitle();
        Assert.assertEquals("My Account – Alchemy LMS", sTitle);
        driver.findElement(By.linkText("Login")).click();
        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();
        WebElement logoutbutton = driver.findElement(By.linkText("Logout"));
        Assert.assertTrue(logoutbutton.isDisplayed());

        driver.findElement(By.linkText("All Courses")).click();
        driver.findElement(By.xpath("//a[text()='See more...'][1]")).click();
        driver.findElement(By.xpath("//*[starts-with(@class,'ld-item-list-item ld-expandable ld-item-lesson-item ld-lesson-item')][1]")).click();
        String coursetitle = driver.findElement(By.xpath("//div[@class='ld-focus-content']//h1")).getText();
        Assert.assertEquals(coursetitle,"Developing Strategy");

        try {
            WebElement markcomplete = driver.findElement(By.xpath("//input[@value='Mark Complete'][1]"));
            if(markcomplete.isDisplayed()) {
                driver.findElement(By.xpath("//input[@value='Mark Complete'][1]")).click();
            }
        }
        catch (Exception  e){
            System.out.println("Course is already completed :"+e);
        }
    }

    @AfterMethod
    public void closebrowser(){
         driver.close();
    }
}

File: Selenium/Project/AlchemyLMSProject.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support import expected_conditions as EC
import unittest

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Initialize the wait object
    wait = WebDriverWait(driver, 10)
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms")
    #Activity1
    print("Title: ", driver.title)
    assert driver.title == "Alchemy LMS – An LMS Application"
    #driver.quit()

    #Activity2
    heading = driver.find_element(By.TAG_NAME,"h1")
    print("Learn from Industry Experts: ",heading.text)
    assert heading.text == "Learn from Industry Experts"

    #Activity3
    title = driver.find_element(By.XPATH,"//h3[@class='uagb-ifb-title']")
    print("Title of the first info box.: ",title.text)
    assert title.text == "Actionable Training"


    #Activity5
    driver.find_element(By.LINK_TEXT,"My Account").click()
    print("Title: ",driver.title)
    assert driver.title == "My Account – Alchemy LMS"

    #Activity6
    driver.find_element(By.LINK_TEXT,"Login").click()
    driver.find_element(By.ID,"user_login").send_keys("root")
    driver.find_element(By.ID,"user_pass").send_keys("pa$$w0rd")
    driver.find_element(By.ID,"wp-submit").click()
    logoutbutton=driver.find_element(By.LINK_TEXT,"Logout")
    assert logoutbutton.is_displayed()

    #Activity7
    driver.find_element(By.LINK_TEXT,"All Courses").click()
    sTitles=driver.find_elements(By.XPATH,"//h3[@class='entry-title']")
    print(len(sTitles))
    for sTitle in sTitles:
        print(sTitle.text)

    #Activity8
    driver.find_element(By.LINK_TEXT,"Contact").click()
    driver.find_element(By.ID,"wpforms-8-field_0").send_keys("Test Name")
    driver.find_element(By.ID,"wpforms-8-field_1").send_keys("test@gmail.com")
    driver.find_element(By.ID,"wpforms-8-field_3").send_keys("Test_Subject")
    driver.find_element(By.ID,"wpforms-8-field_2").send_keys("Test conntent")
    driver.find_element(By.XPATH,"//button[@name='wpforms[submit]']").click()
    sConfirmtest=driver.find_element(By.XPATH,"//div[@id='wpforms-confirmation-8']").text
    print(sConfirmtest)
    assert sConfirmtest=="Thanks for contacting us! We will be in touch with you shortly."

    #Activity9
    driver.find_element(By.LINK_TEXT,"All Courses").click();
    driver.find_element(By.XPATH,"//a[text()='See more...'][1]").click();
    driver.find_element(By.XPATH,"//div[@class='ld-item-list-item-preview'][1]").click();
    coursetitle = driver.title
    print(coursetitle)
    assert coursetitle=="Developing Strategy – Alchemy LMS"


    if(driver.find_element(By.XPATH("//input[@value='Mark Complete'][1]")).is_displayed()):
        driver.find_element(By.XPATH("//input[@value='Mark Complete'][1]")).click();


    #driver.quit()
